powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/internal/LambdaMetricsAspect.java

```diff
@@ -83,12 +87,24 @@ private void coldStartSingleMetricIfApplicable(final String awsRequestId,
                                                    final Metrics metrics) {
         if (metrics.captureColdStart()
                 && isColdStart()) {
-                MetricsLogger metricsLogger = new MetricsLogger();
+            MetricsLogger metricsLogger = new MetricsLogger();
+            try {
                 metricsLogger.setNamespace(namespace(metrics));
+            } catch (InvalidNamespaceException e) {
+                throw new RuntimeException("A valid namespace is required, either pass it to the @Metrics annotation or set the environment variable POWERTOOLS_METRICS_NAMESPACE", e);
+            }
+            try {
                 metricsLogger.putMetric("ColdStart", 1, Unit.COUNT);
+            } catch (InvalidMetricException e) {
+                // should not occur
```

Comment:
How about adding sole logs on debug level? Too many times in my live I've seen things that should not occur ;)